code/Test_definitions/geofencing-subscriptions.feature

```diff
@@ -16,77 +16,130 @@ Feature: Camara Geofencing Subscriptions API, vwip - Operations on subscriptions
-    Then the response code is 200
+    Then the response status code is 200
     And the response header "Content-Type" is "application/json"
     And the response header "x-correlator" has same value as the request header "x-correlator"
-    And the response body has an array of items and each item complies with the OAS schema at "/components/schemas/Subscription"
-
-  @geofencing_subscriptions_05_Operation_to_retrieve_subscription_based_on_an_existing_subscription-id
-  Scenario: Get a subscription based on existing subscription-id.
-    Given the path parameter "subscriptionId" is set to the identifier of an existing subscription
-    When the request "retrieveGeofencingSubscription" is sent
-    Then the response code is 200
+    And the response body complies with an array of OAS schema defined at "#/components/schemas/Subscription"
+    And the response body lists all subscriptions belonging to the API consumer for the identified device
+    And the response property "$.config.subscriptionDetail.device" is not present in any of the subscription records
+
+  @geofencing_subscriptions_06_retrieve_empty_list_3legs
+  Scenario: Check no existing subscription is retrieved in list
+    Given the API consumer has no active subscriptions for the device
+    And the header "Authorization" is set to a valid access token which identifies a valid device
+    When the request "retrieveGeofencingSubscriptionList" is sent
+    Then the response status code is 200
     And the response header "Content-Type" is "application/json"
     And the response header "x-correlator" has same value as the request header "x-correlator"
-    And the response body complies with the OAS schema at "/components/schemas/Subscription"
+    And the response body is an empty array
 
-  @geofencing_subscriptions_06_Operation_to_delete_subscription_based_on_an_existing_subscription-id
-  Scenario: Delete a subscription based on existing subscription-id.
-    Given the path parameter "subscriptionId" is set to the identifier of an existing subscription
+  @geofencing_subscriptions_07_Operation_to_delete_subscription_based_on_an_existing_subscription-id
+  Scenario: Delete the subscription with subscriptionId equal to "id"
+    Given the API consumer has an active subscription with "subscriptionId" equal to "id"
     When the request "deleteGeofencingSubscription" is sent
-    Then the response code is 202 or 204
+    And the path parameter "subscriptionId" is set to "id"
+    Then the response status code is 202 or 204
     And the response header "x-correlator" has same value as the request header "x-correlator"
-    And if the response property $.status is 204 then response body is not available
-    And if the response property $.status is 202 then response body complies with the OAS schema at "/components/schemas/SubscriptionAsync"
+    And if the response property "$.status" is 204 then response body is not present
```

Comment:
The previous steps were the ones in the artifact:

    And if the response property "$.status" is 204 then the response body is not available
    And if the response property "$.status" is 202 then the response body complies with the OAS schema at "#/components/schemas/SubscriptionAsync"